relayer/relays/beacon/header/syncer/api.go

```diff
@@ -86,16 +85,55 @@ type BeaconHeader struct {
 	BodyRoot      common.Hash
 }
 
-func (b *BeaconClient) GetFinalizedHeader() (BeaconHeader, error) {
-	return b.GetHeader("finalized")
-}
+func (b *BeaconClient) GetHeaderBySlot(slot uint64) (BeaconHeader, error) {
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/eth/v1/beacon/headers/%d", b.endpoint, slot), nil)
+	if err != nil {
+		return BeaconHeader{}, fmt.Errorf("%s: %w", ConstructRequestErrorMessage, err)
+	}
+
+	req.Header.Set("accept", "application/json")
+	res, err := b.httpClient.Do(req)
+	if err != nil {
+		return BeaconHeader{}, fmt.Errorf("%s: %w", DoHTTPRequestErrorMessage, err)
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return BeaconHeader{}, fmt.Errorf("%s: %d", HTTPStatusNotOKErrorMessage, res.StatusCode)
+	}
+
+	bodyBytes, err := io.ReadAll(res.Body)
+	if err != nil {
+		return BeaconHeader{}, fmt.Errorf("%s: %w", ReadResponseBodyErrorMessage, err)
+	}
+
+	var response BeaconHeaderResponse
 
-func (b *BeaconClient) GetHeadHeader() (BeaconHeader, error) {
-	return b.GetHeader("head")
+	err = json.Unmarshal(bodyBytes, &response)
+	if err != nil {
+		return BeaconHeader{}, fmt.Errorf("%s: %w", UnmarshalBodyErrorMessage, err)
+	}
+
+	slotScale, err := strconv.ParseUint(response.Data.Header.Message.Slot, 10, 64)
```

Comment:
This is a dumb name, whoops! 😄 There is a name collision with `slot` because `slot` is an input parameter to the `GetHeaderBySlot` method. I could probably use the input parameter slot because it should be the same as the slot returned in the API response, but I think using the response value is perhaps a little safer. I'm renaming this to `slotFromResponse`.

Reply:
ah ok, I thought it might have something to do with Scale-encoding but couldn't see the connection